src/EmployeeTracker/api/views.py

```diff
@@ -0,0 +1,170 @@
+                      User.objects.all()]
+                return Response(xo)
+            elif type == 'y':
+                weekDateFromNow = datetime.now().date() - timedelta(days=365)
+                xo = [self.makingFunction(user, user.attendant.filter(date__gte=weekDateFromNow), 'yearly') for user in
+                      User.objects.all()]
+                return Response(xo)
+            else:
+                raise Exception("Determine the period")
+        except Exception as ex:
+            return Response(str(ex))
+
+    @action(['get'], detail=False)
+    def commingToLeavingForHoleTeam(self, request, *args, **kwargs):
+        try:
+            everyEmpWithWorkingHoure = [self.makingFunction(user, user.attendant, 'yearly') for user in
+                                        User.objects.all()]
+            everyEmpWithleavingHoure = [self.makingFunction(user, user.leaves, 'yearly') for user in
+                                        User.objects.all()]
+            sumWorkHour = 0
+            for emp in everyEmpWithWorkingHoure:
+                sumWorkHour += emp["workingHouresyearly"]
+            if sumWorkHour == 0:
+                raise Exception("This Team has no work hour ")
+            sumLeaveHour = 0
+            for emp in everyEmpWithleavingHoure:
+                sumLeaveHour += emp["workingHouresyearly"]
+
+            return Response(f"this team has an leaving of percent {sumLeaveHour // sumWorkHour}")
+
+        except Exception as ex:
+            return Response(str(ex))
+
+    def makingFunction(self, emp, listAttends, t):
+        list = [x for x in listAttends.values('inTime', 'outTime')]
+        outtime = ""
+        for i in range(len(list)):
+            if list[i]['outTime'] != " ":
+                outtime = list[i]['outTime']
+            else:
+                list[i]['outTime'] = outtime
+        workingHoures = sum([subtractTime(t['inTime'], t['outTime']).hour for t in list])
+        return {"EmpName": emp.username, f"workingHoures{t}": workingHoures}
+
+
+
+class CreateAttendantEmployee(CreateAPIView):
+    serializer_class = AttendantSerializer
+    queryset = Attendants.objects.all()
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request, *args, **kwargs):
+        empid = kwargs.get('empId')
+        if empid != ' ':
+            try:
+                user = User.objects.get(pk=empid)
+                if user is None:
+                    raise Exception("User dose't exist")
+                attend = Attendants.objects.filter(Q(emp_id=empid) & Q(date=datetime.now().date()))
+                if attend.count() == 0:
+                    newAttendant = self.serializer_class(data={'emp': empid})
+                    newAttendant.is_valid(raise_exception=True)
+                    newAttendant.save()
+                    return Response({'message:Thank you'})
+
+                elif attend[0].outTime is None:
+                    cloneattend: Attendants = attend.get()
```

Comment:
what dose this `cloneattend: Attendants = attend.get()` exactly do? I think you can just do:
```suggestion
                    cloneattend = attend.first()
```
and why are you calling it cloneattend?